### PR TITLE
chore(lint): lint ts as well as js

### PR DESCRIPTION
**Pull request by filipesilva**
We weren't linting TS files as part of the CI, so a lot of linting errors were going unnoticed.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
     "test": "node tests/runner",
     "mobile_test": "mocha tests/e2e/e2e_workflow.spec.js",
     "test:inspect": "node --inspect --debug-brk tests/runner",
-    "lint": "eslint ."
+    "lint-js": "eslint .",
+    "lint-ts": "tslint \"**/*.ts\" -c tslint.json -e \"**/__path__/*.ts\" -e \"node_modules/**\" -e \"tmp/**\"",
+    "lint": "npm-run-all -c lint-js lint-ts"
```
Comment (hansl) [inline package.json]: If you're running them sequentially, just do `npm run foo bar` and skip a dependency.

File: package.json [hunk 2]
```diff
@@ -12,7 +12,9 @@
     "test": "node tests/runner",
     "mobile_test": "mocha tests/e2e/e2e_workflow.spec.js",
     "test:inspect": "node --inspect --debug-brk tests/runner",
-    "lint": "eslint ."
+    "lint-js": "eslint .",
```
Comment (hansl) [inline package.json]: could you rename those `eslint` and `tslint`?

**Comment by hansl** [comment]
2 nits. Also, travis fails the lint test.

File: package.json (hunk 1 above)
Comment (filipesilva) [inline package.json]: Tried doing that, but it seems to just send the second one as an argument:
```
$ npm run lint-js lint-ts

> angular-cli@1.0.0-beta.11-webpack.2 lint-js D:\work\angular-cli
> eslint . "lint-ts"
```

File: package.json (hunk 2 above)
Comment (filipesilva) [inline package.json]: Done

**Comment by filipesilva** [comment]
The lint errors I'll address tomorrow probably. There are a fair amount of them.

File: tslint.json [hunk 3]
```diff
@@ -49,7 +49,9 @@
       true,
       "ban-keywords",
       "check-format",
-      "allow-trailing-underscore"
+      "allow-leading-underscore",
```
Comment (hansl) [inline tslint.json]: We'll want to allow one or the other, not both, but I'm fine with it for now (I know it's a big one).

**Comment by hansl** [comment]
LGTM.

**Comment by hansl** [comment]
Please rebase and I'll merge when I can.

**Comment by filipesilva** [comment]
Rebased and pushed. Have a quick review and check the linux CI before merging though, stuff might have been broken.

File: tslint.json (hunk 3 above)
Comment (filipesilva) [inline tslint.json]: Done, left `allow-leading-underscore` only.

Comment (hansl) [inline tslint.json]: Perfect, that's what Angular uses.

**Comment by hansl** [comment]
Still LGTM. I'll merge.

**Pull request merged**